sklearn/svm/_base.py

```diff
@@ -216,6 +216,12 @@ def fit(self, X, y, sample_weight=None):
                     "'auto'. Got '{}' instead.".format(self.gamma)
                 )
         else:
+            if self.kernel == 'linear':
+                warnings.warn(
+                    "Setting 'gamma' when using 'linear' kernel may raise a "
+                    "`ValueError` starting in version 1.1 (renaming of 0.26).",
```

Comment:
I would keep the usual cycle: if we start warning in 1.0, we should start erroring in 1.2 (n+0.2).

Reply:
Thanks for the pointer, I wasn't familiar with the usual cycle so I assumed it would be `n+0.1` :) I'll start making changes when a consensus is reached in #19614.